### PR TITLE
feat(editor): implement executions preview via the new executions tab in node view

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
Closes N8N-4788

### CODE REVIEW

File: packages/editor-ui/src/App.vue [hunk 1]
```diff
@@ -9,14 +9,16 @@
...
 				<router-view name="header"></router-view>
 			</div>
-			<div id="sidebar" :class="$style['sidebar']">
+			<div id="sidebar" :class="$style.sidebar">
 				<router-view name="sidebar"></router-view>
 			</div>
-			<div id="content" :class="$style['content']">
-				<router-view />
+			<div id="content" :class="$style.content">
+				<keep-alive include="NodeView" :max="1">
```
Comment (mutdmour) [inline packages/editor-ui/src/App.vue:19]: why is this needed? could this have memory or perf implications?

File: packages/editor-ui/src/Interface.ts [hunk 2]
```diff
@@ -1059,7 +1062,12 @@ export interface IUsersState {
 	users: {[userId: string]: IUser};
 }
 
-export interface IWorkflowsMap {
+export interface IWorkflowsState {
+	currentWorkflowExecutions: IExecutionsSummary[];
+	activeWorkflowExecution: IExecutionsSummary|null;
+	finishedExecutionsCount: number;
+}
+	export interface IWorkflowsMap {
```
Comment (mutdmour) [inline packages/editor-ui/src/Interface.ts:1070]: nitpick - spacing

File: packages/editor-ui/src/components/mixins/executionsHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				label: 'Status unknown',
+				runningTime: '',
+			};
+
+			if (execution.waitTill) {
+				status.name = 'waiting';
+				status.label = 'Waiting';
+			} else if (execution.stoppedAt === undefined) {
+				status.name = 'running';
+				status.label = 'Running';
```
Comment (mutdmour) [inline packages/editor-ui/src/components/mixins/executionsHelpers.ts:45]: i18n everywhere here?

File: packages/editor-ui/src/components/mixins/executionsHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { genericHelpers } from "./genericHelpers";
+
+export interface IExecutionUIData {
+	name: string;
+	label: string;
+	startTime: string;
+	runningTime: string;
+}
+
+export const executionHelpers = mixins(genericHelpers).extend({
```
Comment (mutdmour) [inline packages/editor-ui/src/components/mixins/executionsHelpers.ts:13]: not a fan of adding more mixins...

File: packages/editor-ui/src/components/WorkflowPreview.vue [hunk 5]
```diff
@@ -141,4 +209,20 @@ export default mixins(showMessage).extend({
 	width: 100%;
 	z-index: 9999999;
 }
+
+.spinner {
+	color: var(--color-primary);
+	position: absolute;
+	top: 50% !important;
+  -ms-transform: translateY(-50%);
+  transform: translateY(-50%);
```
Comment (mutdmour) [inline packages/editor-ui/src/components/WorkflowPreview.vue:218]: nitpick - spacing

**Review by mutdmour — COMMENTED** [review]
awesome. looks good overall

File: packages/editor-ui/src/views/canvasHelpers.ts [hunk 6]
```diff
@@ -515,10 +516,19 @@ export const getMidCanvasPosition = (scale: number, offset: XYPosition): XYPosit
...
 
-export const getBackgroundStyles = (scale: number, offsetPosition: XYPosition) => {
+export const getBackgroundStyles = (scale: number, offsetPosition: XYPosition, executionPreview: boolean) => {
 	const squareSize = GRID_SIZE * scale;
 	const dotSize = 1 * scale;
 	const dotPosition = (GRID_SIZE / 2) * scale;
+
+	if (executionPreview) {
+		return {
+			'background-image': 'linear-gradient(135deg, #f9f9fb 25%, #ffffff 25%, #ffffff 50%, #f9f9fb 50%, #f9f9fb 75%, #ffffff 75%, #ffffff 100%)',
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/editor-ui/src/views/canvasHelpers.ts:526]: Good catch. 
Agree, but I think we are going to need new values for this from @maxtkacz 
I used the existing values and it looks like this in dark mode 😵‍💫:
image

File: packages/editor-ui/src/modules/workflows.ts [hunk 7]
```diff
@@ -43,6 +50,78 @@ const module: Module<IWorkflowsState, IRootState> = {
...
+			let activeExecutions = [];
+			let finishedExecutions = [];
+			const requestFilter: IDataObject = { workflowId: context.rootGetters.workflowId };
+
+			if (!context.rootGetters.workflowId) {
+				return [];
+			}
+			try {
+				if (filter.status === ''|| !filter.finished) {
+					activeExecutions = await makeRestApiRequest(
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/editor-ui/src/modules/workflows.ts:66]: Agree. Updated.

File: packages/editor-ui/src/modules/workflows.ts [hunk 8]
```diff
@@ -43,6 +50,78 @@ const module: Module<IWorkflowsState, IRootState> = {
...
+						{ filter: requestFilter },
+					);
+				}
+				if (filter.status === '' || filter.finished) {
+					if (filter.status === 'waiting') {
+						requestFilter.waitTill = true;
+					} else if (filter.status !== '')  {
+						requestFilter.finished = filter.status === 'success';
+					}
+					finishedExecutions = await makeRestApiRequest(
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/editor-ui/src/modules/workflows.ts:79]: Yeah. It's getting caught in ÈxecutionsView` component to display error if something fails.

File: packages/editor-ui/src/modules/workflows.ts [hunk 9]
```diff
@@ -43,6 +50,78 @@ const module: Module<IWorkflowsState, IRootState> = {
...
+						{
+							filter: requestFilter,
+						},
+					);
+				}
+				context.commit('setTotalFinishedExecutionsCount', finishedExecutions.count);
+				return [...activeExecutions, ...finishedExecutions.results || []];
+			} catch (error) {
+				throw(error);
+			}
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/editor-ui/src/modules/workflows.ts:92]: Same as above.

File: packages/editor-ui/src/App.vue (hunk 1 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/App.vue:19]: This is keeping the main `NodeView` active, so it's not reloading every time we switch tabs. I guess this is the UX we are after in this feature. 
`include` and `max` parameters will limit router view so it only caches one instance of `NodeView`and not other components that render in that slot.

File: packages/editor-ui/src/components/mixins/executionsHelpers.ts (hunk 4 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/components/mixins/executionsHelpers.ts:13]: Me neither, but this logic is used in different components in executions view, so decided to externalize it like this.

File: packages/editor-ui/src/components/mixins/executionsHelpers.ts (hunk 3 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/components/mixins/executionsHelpers.ts:45]: Nope. Forgot about it, will update.

File: packages/editor-ui/src/views/canvasHelpers.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/views/canvasHelpers.ts:526]: that's not too bad

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.200.0`
